src/FSharp.Control.TaskSeq.Test/TaskSeq.TakeWhile.Tests.fs

```diff
@@ -0,0 +1,219 @@
+module TaskSeq.Tests.TakeWhile
+
+open System
+open Xunit
+open FsUnit.Xunit
+open FsToolkit.ErrorHandling
+
+open FSharp.Control
+
+//
+// TaskSeq.takeWhile
+// TaskSeq.takeWhileAsync
+// TaskSeq.takeWhileInclusive
+// TaskSeq.takeWhileInclusiveAsync
+//
+
+module EmptySeq =
+    [<Theory; ClassData(typeof<TestEmptyVariants>)>]
+    let ``TaskSeq-takeWhile has no effect`` variant =
+        Gen.getEmptyVariant variant
+        |> TaskSeq.takeWhile ((=) 12)
+        |> TaskSeq.toListAsync
+        |> Task.map (List.isEmpty >> should be True)
+
+    [<Theory; ClassData(typeof<TestEmptyVariants>)>]
+    let ``TaskSeq-takeWhileAsync has no effect`` variant =
+        Gen.getEmptyVariant variant
+        |> TaskSeq.takeWhileAsync (fun x -> task { return x = 12 })
+        |> TaskSeq.toListAsync
+        |> Task.map (List.isEmpty >> should be True)
+
+// The primary requirement is that items after the item failing the predicate must be excluded
+module FiltersAfterFail =
+    [<Theory; InlineData false; InlineData true>]
+    let ``TaskSeq-takeWhile(Inclusive)? excludes all items after predicate fails`` inclusive =
+        // The only real difference in semantics between the base and the *Inclusive variant lies in whether the final item is returned
+        // NOTE the semantics are very clear on only propagating a single failing item in the inclusive case
+        let f, expected =
+            if inclusive then TaskSeq.takeWhileInclusive, "ABBC"
+            else TaskSeq.takeWhile, "ABB"
```

Comment:
I'm also going to make a few naming changes here (that are kinder to different reporting tools) and lift the `expected` test.

Reply:
Done